railties/lib/rails/generators/app_base.rb

```diff
@@ -140,6 +141,70 @@ def build(meth, *args) # :doc:
         builder.public_send(meth, *args) if builder.respond_to?(meth)
       end
 
+      def deduce_implied_options(options, option_reasons, meta_options)
+        active = options.transform_values { |value| [] if value }.compact
+        irrevocable = (active.keys - meta_options).to_set
+
+        deduction_order = TSort.tsort(
+          ->(&block) { option_reasons.each_key(&block) },
+          ->(key, &block) { option_reasons[key]&.each(&block) }
+        )
+
+        deduction_order.each do |name|
+          reasons = option_reasons[name]&.select(&active).presence
+          active[name] ||= reasons if reasons
+          irrevocable << name if reasons&.any?(irrevocable)
+        end
+
+        revoked = options.select { |name, value| value == false }.keys.to_set - irrevocable
+        deduction_order.reverse_each do |name|
+          revoked += option_reasons[name].to_a if revoked.include?(name)
+        end
+        revoked -= meta_options
+
+        active.filter_map do |name, reasons|
+          reasons -= revoked.to_a
+          [name, reasons] unless revoked.include?(name) || reasons.empty?
+        end.to_h
+      end
+
+      OPTION_IMPLICATIONS = { # :nodoc:
+        (:skip_active_job)     => [:skip_action_mailer, :skip_active_storage],
+        (:skip_active_record)  => [:skip_active_storage],
+        (:skip_active_storage) => [:skip_action_mailbox, :skip_action_text],
+        (:skip_javascript)     => [:skip_hotwire],
+      }
+
+      def imply_options(option_implications = OPTION_IMPLICATIONS, meta_options: [])
```

Comment:
What `meta_options` mean here? Isn't `api` a meta options? Why?

Reply:
A "meta" option's only purpose is to trigger other options.  It has no other effect.  Therefore, meta options should be transparent / ignored when revoking implied options.

For example, both `--skip-active-job` and `--minimal` imply `--skip-active-storage`.  With `--skip-active-job`, `--no-skip-active-storage` is an error, but with `--minimal`, `--no-skip-active-storage` will "undo" `--skip-active-job`.

`--api` is not a meta option because it affects other things besides implying `--skip-asset-pipeline` and `--skip-javascript`.  (And so `--api --no-skip-asset-pipeline` will raise an error.)

Reply:
👍🏽